risc0/zkp/src/core/hash/poseidon/mod.rs

```diff
@@ -15,6 +15,8 @@
 //! An implementation of Poseidon targeting the Baby Bear field with a security
 //! of 128 bits.
 
+extern crate alloc;
```

Comment:
Why do we need these? If we do need it, it should be sufficient to specify it once at the top-level in `lib.rs`

Reply:
Currently `cargo clippy` thinks the crate is unsued when specified in `mod.rs`, so on first pass I made this change, but looking at it again it makes more sense just to turn off the warning.